pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+
+      return r;
+    }
+
+  }
+
+  private static class RunPositionRowMapper implements RowMapper<MisoRunPosition> {
+
+    @Override
+    public MisoRunPosition mapRow(ResultSet rs, int rowNum) throws SQLException {
+      MisoRunPosition p = new MisoRunPosition();
+
+      p.setPosition(rs.getInt("partitionNumber"));
+      p.setRunId(rs.getInt("Run_runId"));
+      p.setPartitionId(rs.getInt("partitionId"));
+      p.setPoolName(rs.getString("pool_name"));
+      p.setPoolDescription(rs.getString("pool_description"));
+      p.setPoolBarcode(rs.getString("pool_barcode"));
+      p.setPoolCreatedById(rs.getInt("pool_createdById"));
+      p.setPoolCreated(rs.getDate("pool_created"));
+
+      return p;
+    }
+
+  }
+
+  public static class SampleRowMapper implements RowMapper<Sample> {
+
+    private static final String SAMPLE_STATUS_READY = "Ready";
+    private static final String SAMPLE_STATUS_NOT_READY = "Not Ready";
+
+    @Override
+    public Sample mapRow(ResultSet rs, int rowNum) throws SQLException {
+      Sample s = new DefaultSample();
+
+      s.setName(rs.getString("name"));
+      s.setDescription(rs.getString("description"));
+      s.setId(rs.getString("id"));
+      String parentId = rs.getString("parentId");
+      if (parentId != null) {
+        Set<String> parents = new HashSet<>();
+        parents.add(parentId);
+        s.setParents(parents);
+      }
+      String misoType = rs.getString("miso_type");
+      if ("Sample".equals(misoType)) {
+        s.setSampleType(SampleTypeConverter.getSampleType(rs.getString("sampleType")));
+      } else {
+        s.setSampleType(SampleTypeConverter.getNonSampleSampleType(misoType, rs.getString("sampleType_platform"),
+            rs.getString("sampleType_description")));
+      }
+      s.setTissueType(rs.getString("tissueType"));
+      s.setProject(rs.getString("project"));
+      s.setArchived(rs.getBoolean("archived"));
+      s.setCreated(rs.getTimestamp("created"));
+      s.setCreatedById(rs.getInt("createdById"));
+      s.setModified(rs.getTimestamp("modified"));
+      s.setModifiedById(rs.getInt("modifiedById"));
+      s.setTubeBarcode(rs.getString("tubeBarcode"));
+      s.setVolume(rs.getFloat("volume"));
+      if (rs.wasNull()) s.setVolume(null);
+      s.setConcentration(rs.getFloat("concentration"));
+      if (rs.wasNull()) s.setConcentration(null);
+      s.setStorageLocation(extractStorageLocation(rs));
+      PreparationKit kit = new DefaultPreparationKit();
+      kit.setName(rs.getString("kitName"));
+      kit.setDescription(rs.getString("kitDescription"));
+      if (kit.getName() != null || kit.getDescription() != null) {
+        s.setPreparationKit(kit);
+      }
+      Set<Attribute> atts = new HashSet<>();
+      for (AttributeKey possibleAtt : AttributeKey.values()) {
+        Attribute att = possibleAtt.extractAttributeFrom(rs);
+        if (att != null) {
+          atts.add(att);
+        }
+      }
+      if (atts.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)